### ISSUE TITLE
What should version 1.0.0 (stable) look like?

### ISSUE THREAD

**Issue by paragonie-scott** [jedisct1/libsodium-php#52]
This is a bit of an open-ended question, and I'd be interested from other libsodium users.

I know libsodium itself is going to have `crypto_pwhash()` when Argon2 is finalized. There are other efforts (platform optimization) which might not surface until a later 1.0.x or 1.1.x.

How many of these are super important to the PHP extension? 

It's possible that PHP 7.1 will come with Argon2 for `password_hash()`/`password_verify()` even if it's not on by default. (I'd ask @ircmaxell to be sure.)

Personally, the only features I've run into that weren't already exposed as of PECL 0.2.1 were the `crypto_auth` API and getting an Ed25519 public key from an Ed25519 secret key. Both of these will be in the next release (PECL 0.2.2, I imagine, but maybe a stable 1.0.0 is next?).

But my use cases are not your use cases. Is there anything else that needs to be added? (I'll do what I can to help make it happen.)

**Comment by jedisct1** [comment]
Documentation.

The current libsodium-php documentation is terrible, if that can be called documentation at all. It's okay for people already familiar with libsodium, but it's clearly not enough for people who are not.

**Comment by paragonie-scott** [comment]
Sure. I can tackle that as a separate project if you'd like. I've set up a repository [here](https://github.com/paragonie/pecl-libsodium-doc) and will be pushing the first real commit in a few hours. (For now, I'm declaring this initiative unofficial.)

I already have a skeleton in place (along with an online reader at paragonie.com) from an PHP 7 ebook.

**Comment by bitbeans** [comment]
@paragonie-scott :+1:

**Comment by paragonie-scott** [comment]
I got the first draft release of that documentation effort finished. When writing out the documentation, I kept my eye out for deficiencies in the PHP bindings that were satisfied in libsodium and I'm fairly confident that they're all tackled.

Which is to say: if you wanted to tag the next release as 1.0.0, I wouldn't object.

**Comment by jedisct1** [comment]
Awesome documentation, Scott!

Alright, the time has come to mark the extension stable, then :)

**Comment by jedisct1** [comment]
Shall I just add a link to your online documentation?

Or can the `pecl-libsodium-doc` chapters be merged to `pecl-libsodium`?

**Comment by paragonie-scott** [comment]
Whichever you prefer. I used relative URLs (and had already programmed my ebook reader to work with those), so long as you place them all in the same directory, the links to other chapters will remain intact.

For licensing purposes: I released it under creative commons, and additionally it could be argued that it was already a remix of the readme, so if you wanted to re-release it under MIT/BSD/ISC, that shouldn't be a thorny issue. (If I'm wrong here, let me know what paperwork I would need to send to make it not thorny.)

If it's easier to link to just add a link, we're not going away any time soon, and our website is archivebot friendly if we ever do.
